tex/generic/pgf/utilities/pgfcalendar.code.tex

```diff
@@ -797,5 +802,207 @@
   \fi}%
 }
 
+%
+% more tests
+%
+\pgfqkeys{/pgf/calendar}{
+  Jan/.code={\ifnum\pgfcalendarifdatemonth=1\pgfcalendarmatchestrue\fi},Jan/.value forbidden,
+  Feb/.code={\ifnum\pgfcalendarifdatemonth=2\pgfcalendarmatchestrue\fi},Feb/.value forbidden,
+  Mar/.code={\ifnum\pgfcalendarifdatemonth=3\pgfcalendarmatchestrue\fi},Mar/.value forbidden,
+  Apr/.code={\ifnum\pgfcalendarifdatemonth=4\pgfcalendarmatchestrue\fi},Apr/.value forbidden,
+  May/.code={\ifnum\pgfcalendarifdatemonth=5\pgfcalendarmatchestrue\fi},May/.value forbidden,
+  Jun/.code={\ifnum\pgfcalendarifdatemonth=6\pgfcalendarmatchestrue\fi},Jun/.value forbidden,
+  Jul/.code={\ifnum\pgfcalendarifdatemonth=7\pgfcalendarmatchestrue\fi},Jul/.value forbidden,
+  Aug/.code={\ifnum\pgfcalendarifdatemonth=8\pgfcalendarmatchestrue\fi},Aug/.value forbidden,
+  Sep/.code={\ifnum\pgfcalendarifdatemonth=9\pgfcalendarmatchestrue\fi},Sep/.value forbidden,
+  Oct/.code={\ifnum\pgfcalendarifdatemonth=10\pgfcalendarmatchestrue\fi},Oct/.value forbidden,
+  Nov/.code={\ifnum\pgfcalendarifdatemonth=11\pgfcalendarmatchestrue\fi},Nov/.value forbidden,
+  Dec/.code={\ifnum\pgfcalendarifdatemonth=12\pgfcalendarmatchestrue\fi},Dec/.value forbidden,
+  leap year/.code={%
+    \pgfutil@tempcnta=#1\relax
```

Comment:
Did you know about [`pgfint.code.tex`](https://github.com/pgf-tikz/pgf/blob/master/tex/generic/pgf/math/pgfint.code.tex)? These macros might make that code a lot easier to read and write.